web-animations-1/Overview.bs

```diff
@@ -4788,11 +4796,17 @@ interface KeyframeEffect : AnimationEffect {
 <div class="attributes">
 
 :   <dfn attribute for=KeyframeEffect>target</dfn>
-::  The element or pseudo-element being animated by this object.
-    This may be <code>null</code> for animations that do not target
+::  The <a>target element</a> being animated by this object
+    (either the <a>effect target</a> if it is an element or its
+    <a>originating element</a> if it is a pseudo-element).
+    This may be `null` for animations that do not target
     a specific element such as an animation that produces a sound
     using an audio API.
 
+:   <dfn attribute for=KeyframeEffect>pseudoElement</dfn>
+::  The <a>effect target</a>'s <a>pseudo-element</a> selector if the effect target is a pseudo-element.
+    This is `null` if the <a>effect target</a> is an element or has no specific target.
```

Comment:
Does this need to define that this is `::before` or `::after` or `::marker`? i.e. both the specific syntax used (not the old-style `:before` etc.) and the set of pseudo elements supported?

Reply:
My idea was that this since there's now the notion of a target Element to hang the Animation off, we could apply animations to all valid pseudo-elements: both the tree-abiding ones, ones that act as alternate style slots (such as ::first-line, ::selection, only supporting the properties those selectors support), and shadow part styles. I see the restriction to before after and marker as an accidental effect of needing a layout object as the effect target.

Reply:
That sounds like a separate issue to this PR which requires further discussion. (I know Gecko has no intention of implementing animations on a wider range of pseudo-elements at this stage.)

For now we still need to define / link to the appropriate syntax here.

Reply:
Are you saying we should add a restriction of which pseudo-elements to support? Currently there is no such restriction in the spec other than the fact that CSSPseudoElement:EventTarget is too much of a mess to even come close to implementing in all but those cases.

Specifying that is returns pseudo-elements in the modern syntax is a good idea.

Reply:
No, no need to an explicit restriction.